R/opendp/src/convert_elements.c

```diff
@@ -0,0 +1,451 @@
+#include <R.h>
+#include <Rmath.h>
+#include <R_ext/Boolean.h>
+#include <R_ext/Rdynload.h>
+#include <Rdefines.h>
+#include <Rinternals.h>
+#include <R_ext/Complex.h>
+#include <R_ext/Callbacks.h>
+
+// Import C headers for rust API
+#include "Ropendp.h"
+#include "opendp.h"
+#include "convert.h"
+
+// TRANSFORMATION
+#define Check_AnyTransformation_Ptr(s)                \
+  do                                                  \
+  {                                                   \
+    if (TYPEOF(s) != EXTPTRSXP ||                     \
+        R_ExternalPtrTag(s) != AnyTransformation_tag) \
+      error("bad Transformation object");             \
+  } while (0)
+
+AnyTransformation *sexp_to_anytransformationptr(SEXP value)
+{
+  PROTECT(value);
+
+  int errorOccurred;
+  SEXP class_expr = lang2(install("class"), value);
+  SEXP class = R_tryEval(class_expr, R_GlobalEnv, &errorOccurred);
+  if (errorOccurred)
+    error("could not determine class");
+
+  if (str_equal(sexp_to_charptr(class), "transformation"))
+  {
+    SEXP call = PROTECT(Rf_lang2(value, PROTECT(Rf_mkString("ptr"))));
+    value = Rf_eval(call, R_GlobalEnv);
+    UNPROTECT(2);
+  }
```

Comment:
Needs an else clause here?

Reply:
Not necessary, as the first condition in the Check_AnyTransformationPtr will catch the case where the class doesn't match.